libs/test/SmartStoreTest/src/com/salesforce/androidsdk/store/SmartStoreLoadTest.java

```diff
@@ -91,237 +83,180 @@ protected void tearDown() throws Exception {
 		super.tearDown();
 	}
 
-    /**
-     * TEST: Upsert 1,2,...,MAX_NUMBER_ENTRIES entries (with just a couple of fields) into a soup
-     * @throws JSONException 
-     */
-    public void testUpsertManyEntries() throws JSONException {
-        Log.i("SmartStoreLoadTest", "In testUpsertManyEntries");
-        for (int k=1; k<MAX_NUMBER_ENTRIES; k*=2) {        
-        	upsertManyEntries(k);
-        }
+
+    //
+    // Tests
+    //
+
+    public void testUpsertQuery1StringIndex1field20characters() throws JSONException {
+        tryUpsertQuery(Type.string, NUMBER_ENTRIES, 1, 20, 1);
```

Comment:
Rewrote most of the tests here. Now you can specify number of indexes / type of index / number of rows / number of fields per soup element / number of characters per field. And timings for upserts and various queries (exact/like/all) will be logged. We should probably run various permutations on actual devices and collect the number.
Locally, I observed json1 and string to be comparable at query time, but json1 to be faster for inserts (as expected).